### ISSUE TITLE
[BUG/ISSUE] Errors when trying to run a nested simulation of AS using GCClassic 13.1.1

### ISSUE THREAD

**Issue by olivian13** [geoschem/geos-chem#788]
### Description of the problem
Hi! I am attempting to run a nested (0.5x0.625) fullchem simulation of AS, but I’ve been running into two issues, both resulting in it failing before timestepping.

The first issue is that it fails when trying to open a specific AFCID file:

![image](https://user-images.githubusercontent.com/49542531/126679220-d0732d7e-2d6a-4694-b151-1135ed4066d0.png)
(See log_withAFCID below).

The second issue is associated with my attempt at troubleshooting where I tried turning off the AFCID emissions in HEMCO_Config.rc. However, I also ran into the same issue when I tried using a smaller lat-lon range for the AS simulation (which results in that specific AFCID file no longer being read in). In these cases, the boundary condition and restart files are read in incorrectly, reporting all values as **************. It fails citing an error in computing F_OF_PBL (see log_noAFCID below).

### Description of troubleshooting performed
A nested run for NA was also tested. For this region, the restart and boundary condition files were read in successfully and it started timestepping. Also, the same boundary condition and restart files for the AS region were tested in v13.0.0 with the AFCID emissions still on and that ran successfully.

### GEOS-Chem version
v13.1.1

### Description of modifications
The Luo et al., wet scavenging updates have been implemented into this version of GC as mentioned by jdhask in #522 (available [here]( https://github.com/jdhask/geos-chem/tree/feature/gan-luo_wetdep)). However, I also ran the same coarse and nested runs in v13.1.2 (with no modifications) and it resulted in the same errors.

### Log files
- Build log (if applicable):
- Run logs (if applicable): 
[log_withAFCID.txt](https://github.com/geoschem/geos-chem/files/6863962/log_withAFCID.txt)
[log_noAFCID.txt](https://github.com/geoschem/geos-chem/files/6863961/log_noAFCID.txt)
- Run script (if applicable):

### Software versions
- CMake version:
- Compilers (Intel or GNU, and version):
- NetCDF version:

**Comment by yantosca** [comment]
Thanks for writing @olivian13.  We are currently validating the Luo wetdep (which will go into 13.2.0) with benchmark simulations.  You might want to wait until 13.2.0 ships in August.  It is possible that the implementation of the Luo wetdep in your code may have some unforeseen bugs or issues that would have been caught in our testing.  

The `failed in xmap_r4r4` error is a regridding issue, but it could be a side-effect caused by something else like an out-of-bounds error.  Have you tried building GEOS-Chem from scratch with the debug option.  Navigate to your run directory and type:

```
mkdir debug
cd debug
cmake ../CodeDir -DCMAKE_BUILD_TYPE=Debug -DRUNDIR=..
make -j
make install
cd ..
```
and then try to run GEOS-Chem again.  If it fails, and you get a core file, then you can use the GDB debugger (which usually comes with most Linux installs) to find out exactly where the run died:
```
gdb gcclassic core.XXXX
```

**Comment by olivian13** [comment]
Hi @yantosca, thanks for getting back to me! 

To make things less complicated, I am now running all of my test runs in v13.1.2 without the Luo wetdep update. I still get the same errors that are seen in the above log files, meaning that wetdep implementation is not causing these errors.

I tried building GEOS-Chem with the debug option on:
<img width="466" alt="cmake (2)" src="https://user-images.githubusercontent.com/49542531/126820095-a6645a86-2d28-4892-8ef8-9c8dd02d9833.PNG">

No core file is produced though when the run fails, so I'm still unsure exactly where the run is dying when the `failed in xmap_r4r4` error is occurring. I do know that turning off the AFCID emissions prevents this error from occurring. With the debugging option still on and the AFCID emissions off, this error message is produced:

```
forrtl: error (65): floating invalid
```

Here are the associated log files: 
[HEMCO.log](https://github.com/geoschem/geos-chem/files/6871858/HEMCO.log)
[log_noAFCID_debug.txt](https://github.com/geoschem/geos-chem/files/6871859/log_noAFCID_debug.txt)

I see on the wiki that using safe division can help to resolve this error, but with no core file that I can find, I'm a little unsure exactly where the safe division is required.

The boundary condition and restart files that were generated and being used in v13.1.2 were also tested in v13.0.0 (with AFCID on) and it ran successfully in that version. The log file for the v13.0.0 run is here: [log_v13_0_0.txt](https://github.com/geoschem/geos-chem/files/6871864/log_v13_0_0.txt)  (I stopped the run early after it reached timestepping)

Even though the nested runs in both versions are set to have the same lat-lon range (lat: 60 to 150, lon: -11 to 55), the grids reported in the log files are different and I'm not sure what is the causing the difference. The strangest part being that in the v13.1.2 log the longitude edges have this range: 59.688,   60.312, ...., 148.438,  149.062, -90.312. Could this be related to the errors I'm encountering?

**Comment by lizziel** [comment]
I was able to reproduce this issue, also reported in https://github.com/geoschem/geos-chem/issues/798. I'll investigate and report back what I find. It indeed seems to be a bug.

**Comment by yantosca** [comment]
@olivian13 , please see [this comment(https://github.com/geoschem/geos-chem/issues/798#issuecomment-890026756) at issue #798 for the solution.

**Comment by yantosca** [comment]
I have confirmed that the fix mentioned above is able to get past the error in the nested-grid simulation.  I will close out this issue.  The fix is available in our `patch/13.2.1` branch at this repository.  This will ship with version 13.2.1 (which will be issued simultaneously with 13.2.0, once 13.2.0 passes benchmarking).   Thanks again for bringing this to our attention!
